middlewares/auth.js

```diff
@@ -18,3 +18,29 @@ module.exports.getUserInfo = (req, res) => {
   })
   .catch(error => res.send(`Access Token Error ${error.message}`));
 };
+
+const interceptNonStaff = (req, res, next) => {
+  const accessToken = req.session.token && req.session.token.access_token;
+  const options = {
+    method: 'GET',
+    url: `${baseUrl}:${lmsPort}/api/user/v1/is_staff`,
+    headers: {
+      Authorization: `Bearer ${accessToken}`
+    }
+  };
+
+  axios(options)
```

Comment:
You will need to cache this result in session.
Since this is implemented as middleware, it will be executed on each request by auth user. Thus, it will make an API request to check if the user is staff. By caching it in session, we will need to check is_staff only once per user session.